File: Easy/Anagram/anagram.py
#User function Template for python3


class Solution:
    
    #Function is to check whether two strings are anagram of each other or not.
    def isAnagram(self,a,b):
        #code here
        n1 = len(a)
        n2 = len(b)
        if(n1 != n2):
            return False
        a = sorted(a)
        b = sorted(b)
        
        for i in range(n1):
            if(a[i] != b[i]):
                return False
        return True


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__=='__main__':
    t = int(input())
    for i in range(t):
        a,b=map(str,input().strip().split())
        if(Solution().isAnagram(a,b)):
            print("YES")
        else:
            print("NO") 
# } Driver Code Ends

File: Difficulty: Easy/Anagram/anagram.java
//{ Driver Code Starts
import java.io.*;
import java.lang.*;
import java.util.*;
import java.util.stream.*;

class GFG {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());
        while (t-- > 0) {
            String s1 = br.readLine(); // first string
            String s2 = br.readLine(); // second string

            Solution obj = new Solution();

            if (obj.areAnagrams(s1, s2)) {
                System.out.println("true");
            } else {
                System.out.println("false");
            }
            System.out.println("~");
        }
    }
}
// } Driver Code Ends




class Solution
{    
    //Function is to check whether two strings are anagram of each other or not.
    public static boolean areAnagrams(String a,String b)
    {
        
        // Your code here
        int[] count = new int[256];
        
        if(a.length() != b.length()){
            return false;
        }
        
        for(int i = 0;i<a.length();i++){
            count[a.charAt(i)]++;
            count[b.charAt(i)]--;
        }
        
        for(int i : count){
            if(i != 0){
                return false;
            }
        }
        
        return true;
        
    }
}